Library/Homebrew/requirements/java_requirement.rb

```diff
@@ -6,6 +6,8 @@ class JavaRequirement < Requirement
   download "http://www.oracle.com/technetwork/java/javase/downloads/index.html"
 
   satisfy :build_env => false do
+    return false unless File.executable? "/usr/libexec/java_home"
```

Comment:
See comments above about perhaps checking for `java` and `java_home` in the `$PATH`